pkg/cli/completion_test.go

```diff
@@ -17,6 +17,62 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+func TestProjectPrefixCompletions(t *testing.T) {
+	appNames := []string{"test-1", "acorn-1", "acorn-2", "hub", "test-2"}
+	apps := make([]apiv1.App, 0, len(appNames))
+	for _, name := range appNames {
+		apps = append(apps, apiv1.App{ObjectMeta: metav1.ObjectMeta{Name: name}})
+	}
+
+	mockClientFactory := &testdata.MockClientFactory{
+		AppList: apps,
+		ProjectItem: &apiv1.Project{
+			TypeMeta: metav1.TypeMeta{},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "project",
+			},
+		},
+	}
+	cmd := new(cobra.Command)
+	cmd.SetContext(context.Background())
+
+	tests := []struct {
+		name          string
+		args          []string
+		toComplete    string
+		wantNames     []string
+		wantDirective cobra.ShellCompDirective
+	}{
+		{
+			name:          "Project Prefix Doesn't Exist",
+			toComplete:    "nonexist::tatatat",
+			wantNames:     []string{},
+			wantDirective: cobra.ShellCompDirectiveNoFileComp,
+		},
+		{
```

Comment:
What about a test case where it autocompletes a project without the `::`?

Reply:
I can't write any successful tests because of how this is implemented unfortunately. :(